lib/controllers/ServiceFabrikApiController.js

```diff
@@ -111,19 +111,32 @@ class ServiceFabrikApiController extends BaseController {
     }
     return Promise
       .try(() => {
-        const space_guid = req.body.space_guid || req.query.space_guid;
-        if (space_guid) {
-          this.validateUuid(space_guid, 'Space GUID');
-          return space_guid;
+        /* Following statement to address cross consumption scenario*/
+        const platform = (req.body ? (req.body.context ? req.body.context.platform : undefined) : undefined) ||
+          req.query.platform;
+        if (platform !== undefined) {
+          _.set(req, 'entity.platform', platform);
+        }
+        /*Following statement for backward compatibility*/
+        const tenant_id = req.body.space_guid || req.query.space_guid ||
+          (platform ? (req.query.tenant_id || req.body.context.space_guid || req.body.context.namespace) : undefined);
+        if (tenant_id) {
+          if (!BaseController.uuidPattern.test(tenant_id) &&
```

Comment:
I think this check is for the fact that cloudfoundry would have space_guid in GUID format and K8S will have it as name of the namespace, not guid. But in that case, if the tenant_id is from CF, we dont need to mtch it with k8s and vice versa.